sopel/bot.py

```diff
@@ -141,6 +141,20 @@ def __init__(self, config, daemon=False):
             self.config.core.host_blocks = []
         self.setup()
 
+    @property
+    def hostmask(self):
+        if self.users.contains(self.nick):
+            # If the user exists, the `User` object will provide a hostmask
+            return self.users.get(self.nick).hostmask
+
+        if not self.users:
+            raise KeyError("'hostmask' not available yet: bot must be connected and in at least one channel.")
```

Comment:
The `KeyError` makes no sense to me here. A `KeyError` means that a key is not in an index, a list, or a dict - any kind of data-structure that is act as a container. Here, `self.users` act as a container, and should be treated as such. Can you look at this so it's the other other way arround:

```python
if not self.users:
    raise RuntimeErrror('not available')

if not self.users.contains(self.nick):
    raise KeyError('Not in its user-list')

return self.users.get(self.nick).hostsmask
```

* No user at all: something is wrong (probably)
* Not the bot: KeyError
* Otherwise: the bot's hostsmask (if it's known)

I'd even dare to say that we could squash the first two cases in one: no user at all or no bot in it should throw the exact same `KeyError` - no `RuntimeError` needed, in my opinion.

Reply:
Sounds good. Message for the `KeyError` okay in the changes?

Reply:
Looking good!